build/tasks/node_smoke_tests.js

```diff
@@ -0,0 +1,46 @@
+"use strict";
+
+const fs = require( "fs" );
+const util = require( "util" );
+const exec = util.promisify( require( "child_process" ).exec );
+const verifyNodeVersion = require( "./lib/verifyNodeVersion" );
+const { argv } = require( "process" );
+const args = argv.slice( 2 );
+
+const allowedModules = [ "commonjs", "module" ];
+
+if ( !verifyNodeVersion() ) {
+	return;
+}
+
+// Fire up all tests defined in test/node_smoke_tests/*.js in spawned sub-processes.
+// All the files under test/node_smoke_tests/*.js are supposed to exit with 0 code
+// on success or another one on failure. Spawning in sub-processes is
+// important so that the tests & the main process don't interfere with
+// each other, e.g. so that they don't share the `require` cache.
+
+async function runTests() {
+	const promises = args.map( async( arg ) => {
+		const [ moduleType, module ] = arg.split( ":" );
+		if ( !allowedModules.includes( moduleType ) ) {
+			throw new Error(
+				`Usage: \`node_smoke_tests [${allowedModules.join( "|" )}]:JQUERY\``
+			);
+		}
+		const files = await fs.promises.readdir(
```

Comment:
`fs.promises` is long deprecated, we should use `fs/promises` or `node:fs/promises`. If you check for Node version early enough, this should be fine.

Reply:
I actually didn't know that. Just thought it was an alias.

Reply:
Bah, apparently `fs/promises` is not supported in Node 10.

Reply:
Out of curiosity, can you point me to the deprecation notice? `fs.promises` is no longer in the docs, but I also can't find a notice saying it's deprecated (https://github.com/nodejs/node/blob/main/doc/api/deprecations.md).

Reply:
They may not have had to deprecate formally as `fs.promises` existed while the API was experimental so they were free to do any changes. I wouldn't depend on undocumented APIs which it is.

It shouldn't matter that `fs/promises` is not supported in Node 10 as we already don't run Node smoke tests on Node 10 on `main`. From my comment above:
> If you check for Node version early enough, this should be fine.

One of the advantages of CommonJS vs. ESM. 😉

Reply:
Yes, but I'm also using `fs.promises` in the new build script, which does run on Node 10. I'd say let's switch all of them once we can drop Node 10 (and Jenkins), which hopefully will be soon.

Reply:
OK, fair enough.